python/lib/dependabot/python/update_checker.rb

```diff
@@ -234,8 +234,8 @@ def updated_version_req_lower_bound
                       .reject { |req_string| req_string.start_with?("<") }
                       .select { |req_string| req_string.match?(VERSION_REGEX) }
                       .map { |req_string| req_string.match(VERSION_REGEX) }
-                      .select { |version| Gem::Version.correct?(version) }
-                      .max_by { |version| Gem::Version.new(version) }
+                      .select { |version| Python::Version.correct?(version) }
```

Comment:
We need to use th correct method that is ecosystem specific and not the one from gem version that checks against a completely different regex.